src/game/common/audio/audioeventinfo.h

```diff
@@ -121,4 +123,72 @@ class AudioEventInfo : public MemoryPoolObject
 
 private:
     static FieldParse s_audioEventParseTable[];
+    friend class AudioEventRTS;
 };
+
+class DynamicAudioEventInfo : public AudioEventInfo
+{
+    IMPLEMENT_POOL(DynamicAudioEventInfo);
+
+    void *operator new(size_t size, void *dst) { return dst; }
+    void operator delete(void *p, void *q) {}
+
+public:
+    enum AudioOverrideFlags
+    {
+        OVERRIDE_AUDIO_NAME,
+        OVERRIDE_LOOP_FLAG,
+        OVERRIDE_LOOP_COUNT,
+        OVERRIDE_VOLUME,
+        OVERRIDE_MIN_VOLUME,
+        OVERRIDE_MIN_RANGE,
+        OVERRIDE_MAX_RANGE,
+        OVERRIDE_PRIORITY,
+        OVERRIDE_COUNT
+    };
+
+#ifdef GAME_DLL
+    DynamicAudioEventInfo *Hook_Ctor() { return new (this) DynamicAudioEventInfo(); }
+    DynamicAudioEventInfo *Hook_Ctor2(const AudioEventInfo &audio) { return new (this) DynamicAudioEventInfo(audio); }
+#endif
+
+    DynamicAudioEventInfo() {}
+    DynamicAudioEventInfo(const AudioEventInfo &audio) : AudioEventInfo(audio) {}
+
+    virtual ~DynamicAudioEventInfo() override {}
+    virtual bool Is_Level_Specific() override { return true; }
+    virtual DynamicAudioEventInfo *Get_Dynamic_Event_Info() override { return this; }
+    virtual const DynamicAudioEventInfo *Get_Dynamic_Event_Info() const override { return this; }
+
+    bool Name_Overridden() { return m_overrideFlags.Test(OVERRIDE_AUDIO_NAME); }
+    bool Loop_Flag_Overridden() { return m_overrideFlags.Test(OVERRIDE_LOOP_FLAG); }
+    bool Loop_Count_Overridden() { return m_overrideFlags.Test(OVERRIDE_LOOP_COUNT); }
+    bool Volume_Overridden() { return m_overrideFlags.Test(OVERRIDE_VOLUME); }
+    bool Min_Volume_Overridden() { return m_overrideFlags.Test(OVERRIDE_MIN_VOLUME); }
+    bool Max_Range_Overridden() { return m_overrideFlags.Test(OVERRIDE_MAX_RANGE); }
+    bool Min_Range_Overridden() { return m_overrideFlags.Test(OVERRIDE_MIN_RANGE); }
+    bool Priority_Overridden() { return m_overrideFlags.Test(OVERRIDE_PRIORITY); }
+
+    Utf8String &Get_Original_Name()
```

Comment:
Returns class member with mutable access. Is this proper?

Reply:
Yes this is how the original looks.